### DISCUSSION TITLE
Is `@astrojs/image` Planning On Becoming Framework Agnostic?

### DISCUSSION THREAD

**Discussion by neelsudhakaran** [withastro/roadmap#383] [Proposal]
It's been great having an official Astro Image component, but it would be even better to have access to the same Image API in any supported framework components.

Part 4 of the [original `@astrojs/image` RFC](https://github.com/withastro/astro/issues/492) even discusses this—outlining the following solution:

```ts
import { Image } from 'astro/react'; // React component
import { Image } from 'astro/svelte'; // Svelte component
import { Image } from 'astro/web'; // Web component
// Etc...
```

Given the current state of `@astro/image`, I think a similar model can still work just as follows instead:

```ts
import { Image } from '@astrojs/image/components' // Astro Component
import { Image } from '@astrojs/image/components/[framework]' // [Framework] Component
```

I switch around from using Astro to using Framework components a lot, and this has been (by far) the most annoying part of making the swap. I'd love to know if anyone else feels the same way!

_**PS:** I'm open to helping out and making this possible (via a formal RFC) if that seems like the best path forward!_

**Comment by mandar1jn** [comment]
There is already a getImage component so this should be doable

**Comment by panwauu** [comment]
I think this would be great and make the image integration more user friendly!

one concern is to keep the component behavior consistent between frameworks. 

Maybe for tests we could statically render the component in every framework and compare it to the same output as they should be the same. Just thinking here… 🤔

**Reply by neelsudhakaran** [reply]
I think that's a great idea!

**Comment by mandar1jn** [comment]
I’m currently working on adding react support. It seemed like a good first real Astro pr. struggling with getImage being async. React seems to never actually resolve it

**Reply by neelsudhakaran** [reply]
My exact same issue when trying this out! I'm trying to see if there is a workaround to that...

**Reply by neelsudhakaran** [reply]
Do you think something like this would work to avoid the async headache?

```tsx
export const Image: FC<Props> = function({ loading = 'lazy', decoding = 'async', ...props }) {
	
	if (props.alt === undefined || props.alt === null) {
		warnForMissingAlt();
	}
	
	async function setup() {
		const attrs = await getImage(props);
	}

	setup()

	return <img {...attrs} loading={loading} deconding={decoding} />

}
```

**Reply by mandar1jn** [reply]
> Do you think something like this would work to avoid the async headache?
> 
> ```tsx
> export const Image: FC<Props> = function({ loading = 'lazy', decoding = 'async', ...props }) {
> 	
> 	if (props.alt === undefined || props.alt === null) {
> 		warnForMissingAlt();
> 	}
> 	
> 	async function setup() {
> 		const attrs = await getImage(props);
> 	}
> 
> 	setup()
> 
> 	return <img {...attrs} loading={loading} deconding={decoding} />
> 
> }
> ```

I just found something like that too. It’s late here but I’ll implement it tomorrow. The rest of the object is already done.

**Comment by samijaber** [comment]
Mitosis is the perfect tool for this: https://github.com/builderio/mitosis . Would probably be delightfully easy to implement and publish, given that the `getImage` code is already framework agnostic.

I am the core maintainer of Mitosis, so if anyone digs into making `@astrojs/image` framework agnostic using Mitosis, shoot me a DM on the Mitosis discord or twitter with any questions and I'll do my best to help out.

**Reply by jasikpark** [reply]
I think someone already attempted once in #integrations, but they were stumped by how to call the async getImage() function on server render

**Reply by samijaber** [reply]
Ah, could you please tag me (or tell them to tag me) in the appropriate channel/place and I will try to debug? There is nothing about Mitosis that should make `async` code fundamentally hard to write, so I'm sure I can lend a hand.

My username is `sami#2705`

**Comment by delucis** [comment]
Closing as `@astrojs/image` is deprecated with image functionality moved inside the `astro` package instead.
